meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-bsp-common.inc

```diff
@@ -11,14 +11,33 @@ deploy_nxp_imx() {
                 type_idx=$(expr $type_idx + 1);
                 if [ $type_idx -eq $machine_idx ]; then
                     install -d ${DEPLOYDIR}/${BOOT_TOOLS}
-                    # When sign is enabled the final DTB should be copied from deploy dir
-                    if [ "${UBOOT_SIGN_ENABLE}" = "1" ]; then
-                        install -m 0644 ${DEPLOY_DIR_IMAGE}/${UBOOT_DTB_IMAGE} ${DEPLOYDIR}/${BOOT_TOOLS}/${UBOOT_DTB_NAME}
-                    else
-                        install -m 0644 ${B}/${config}/arch/arm/dts/${UBOOT_DTB_NAME} ${DEPLOYDIR}/${BOOT_TOOLS}
-                    fi
                     install -m 0644 ${B}/${config}/spl/u-boot-spl-nodtb.bin ${DEPLOYDIR}/${BOOT_TOOLS}/u-boot-spl-nodtb.bin-${MACHINE}-${type}
                     install -m 0644 ${B}/${config}/u-boot-nodtb.bin ${DEPLOYDIR}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}-${type}
+                    UBOOT_DTB_NAME_FLAGS="${type}:${UBOOT_DTB_NAME}"
+                    for key_value in ${UBOOT_DTB_NAME_FLAGS}; do
+                        local type_key="${key_value%%:*}"
+                        local dtb_name="${key_value#*:}"
+                        if [ "$type_key" = "$type" ]
+                        then
+                            bbnote "UBOOT_CONFIG = $type, UBOOT_DTB_NAME = $dtb_name"
+                            # There is only one ${dtb_name}, the first one. All the other are with the type appended
+                            if [ ! -f "${DEPLOYDIR}/${BOOT_TOOLS}/${dtb_name}" ]; then
+                                # When sign is enabled the final DTB should be copied from deploy dir
+                                if [ "${UBOOT_SIGN_ENABLE}" = "1" ]; then
+                                    install -m 0644 ${DEPLOY_DIR_IMAGE}/${UBOOT_DTB_IMAGE}  ${DEPLOYDIR}/${BOOT_TOOLS}/${dtb_name}
+                                else
+                                    install -m 0644 ${B}/${config}/arch/arm/dts/${dtb_name}  ${DEPLOYDIR}/${BOOT_TOOLS}/${dtb_name}
+                                fi
+                            else
+                                bbwarn "Use custom wks.in for $dtb_name = $type"
+                            fi
+                            install -m 0644 ${B}/${config}/arch/arm/dts/${dtb_name}  ${DEPLOYDIR}/${BOOT_TOOLS}/${dtb_name}-${type}
+                        fi
+                        unset type_key
+                        unset dtb_name
+                    done
+
+                    unset UBOOT_DTB_NAME_FLAGS
                 fi
             done
```

Comment:
So far for when UBOOT_CONFIG is empty, I've been using this:

```
+++ b/recipes-bsp/u-boot/u-boot-imx_2023.04.bb
@@ -53,6 +53,10 @@ do_deploy:append:mx8m-generic-bsp() {
             unset  j
         done
         unset  i
+    else
+        install -d ${DEPLOYDIR}/${BOOT_TOOLS}
+        install -m 0644 ${B}/${config}/arch/arm/dts/${UBOOT_DTB_NAME}  ${DEPLOYDIR}/${BOOT_TOOLS}/${UBOOT_DTB_NAME}
+        install -m 0644 ${B}/${config}/u-boot-nodtb.bin                ${DEPLOYDIR}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}
     fi
 }
```

Reply:
Cool, we can follow this up with your other changes in meta-freescale.